code/modules/antagonists/traitor/objectives/demoralise_assault.dm

```diff
@@ -0,0 +1,119 @@
+/datum/traitor_objective/target_player/assault
+	name = "Assault %TARGET% the %JOB TITLE%"
+	description = "%TARGET% has been identified as a potential future agent. \
+		Pick a fight and give them a good beating. \
+		%COUNT% hits should reduce their morale and have them questioning their loyalties. \
+		Try not to kill them just yet, we may want to recruit them in the future."
+
+	abstract_type = /datum/traitor_objective/target_player
+	duplicate_type = /datum/traitor_objective/target_player
+
+	progression_minimum = 0 MINUTES
+	progression_maximum = 30 MINUTES
+	progression_reward = list(4 MINUTES, 8 MINUTES)
+	telecrystal_reward = list(0, 1)
+
+	/// Min attacks required to pass the objective. Picked at random between this and max.
+	var/min_attacks_required = 2
+	/// Max attacks required to pass the objective. Picked at random between this and min.
+	var/max_attacks_required = 5
+	/// The random number picked for the number of required attacks to pass this objective.
+	var/attacks_required = 0
+	/// Total number of successful attacks recorded.
+	var/attacks_inflicted = 0
+
+/datum/traitor_objective/target_player/assault/proc/on_attacked(mob/source, mob/living/attacker, attack_flags)
+	SIGNAL_HANDLER
+
+	// Only care about attacks from the objective's owner.
+	if(attacker != handler.owner.current)
+		return
+
+	// We want some sort of damaging attack to trigger this, rather than shoves and non-lethals.
+	if(!(attack_flags & ATTACKER_DAMAGING_ATTACK))
+		return
+
+	attacks_inflicted++
+
+	if(attacks_inflicted == attacks_required)
+		succeed_objective()
+
+/datum/traitor_objective/target_player/assault/on_objective_taken(mob/user)
+	. = ..()
+
+	target.AddElement(/datum/element/relay_attackers)
```

Comment:
That would be a fix for another PR, since it seems sensible that multiple things in the future may want to be listening for attacks on the same atom. Possibly overriding, or refactoring it into something that keeps its own state so it knows what things it's already hooked into.